source/Calamari.Aws/Integration/AwsScriptWrapper.cs

```diff
@@ -16,7 +19,11 @@ public class AwsScriptWrapper : IScriptWrapper
         readonly ILog log;
         readonly IVariables variables;
         public int Priority => ScriptWrapperPriorities.CloudAuthenticationPriority;
-        bool IScriptWrapper.IsEnabled(ScriptSyntax syntax) => true;
+
+        bool IScriptWrapper.IsEnabled(ScriptSyntax syntax) =>
+            variables.Get(SpecialVariables.Account.AccountType) == "AmazonWebServicesAccount" ||
+            string.Equals(variables.Get(AwsSpecialVariables.Authentication.UseInstanceRole), bool.TrueString,
+                StringComparison.InvariantCultureIgnoreCase);
```

Comment:
This script wrapper would blow up if you didn't have the correct variables set with your AWS Creds (and previously it wouldn't be loaded if you didn't specify the AWS extension) so I had to put some checks in (which mirror those in Server in `WithAwsCalamariExtensionsIfRequired()` method.